src/components/survey/building.tsx

```diff
@@ -1,14 +1,315 @@
+      <TextField disabled select label={ strings.survey.building.buildingClass }>
+        <MenuItem>{ strings.generic.notImplemented }</MenuItem>
+      </TextField>
+      {
+        renderWithDebounceNumberTextField(
+          "constructionYear",
+          strings.survey.building.year,
+          building?.constructionYear || 0,
+          onBuildingPropChange
+        )
+      }
+      {
+        renderWithDebounceNumberTextField(
+          "space",
+          strings.survey.building.area,
+          building?.space || 0,
+          onBuildingPropChange
+        )
+      }
+      {
+        renderWithDebounceNumberTextField(
+          "volume",
+          strings.survey.building.volume,
+          building?.volume || 0,
+          onBuildingPropChange
+        )
+      }
+      {
+        renderWithDebounceNumberTextField(
+          "floors",
+          strings.survey.building.floors,
+          building?.floors || 0,
+          onBuildingPropChange
+        )
+      }
+      {
+        renderWithDebounceNumberTextField(
+          "basements",
+          strings.survey.building.basementFloors,
+          building?.basements || 0,
+          onBuildingPropChange
+        )
+      }
+      {
+        renderWithDebounceTextField(
+          "foundation",
+          strings.survey.building.foundationMaterial,
+          building?.foundation || "",
+          onBuildingPropChange
+        )
+      }
+      {
+        renderWithDebounceTextField(
+          "supportingStructure",
+          strings.survey.building.supportingStructure,
+          building?.supportingStructure || "",
+          onBuildingPropChange
+        )
+      }
+      {
+        renderWithDebounceTextField(
+          "facadeMaterial",
+          strings.survey.building.façadeMaterial,
+          building?.facadeMaterial || "",
+          onBuildingPropChange
+        )
+      }
+      {
+        renderWithDebounceTextField(
+          "roofType",
+          strings.survey.building.roofStructure,
+          building?.roofType || "",
+          onBuildingPropChange
+        )
+      }
+    </>
+  );
+
+  /**
+   * Renders address form
+   */
+  const renderAddressForm = () => (
+    <>
+      <Typography variant="h3" sx={{ marginBottom: 0.5 }}>
+        { strings.survey.building.address }
+      </Typography>
+      {
+        renderWithDebounceTextField(
+          "streetAddress",
+          strings.survey.building.street,
+          building?.address?.streetAddress || "",
+          onBuildingAddressPropChange
+        )
+      }
+      {
+        renderWithDebounceTextField(
+          "city",
+          strings.survey.building.city,
+          building?.address?.city || "",
+          onBuildingAddressPropChange
+        )
+      }
+      {
+        renderWithDebounceTextField(
+          "postCode",
+          strings.survey.building.postalCode,
+          building?.address?.postCode || "",
+          onBuildingAddressPropChange
+        )
+      }
+    </>
+  );
```

Comment:
Check that building and address are not undefined.
Spread needed variables from building.